proposals/NNNN-Removing-URL-Param-Max-Length.md

```diff
@@ -0,0 +1,84 @@
+
+## Proposed solution
+
+We should remove the max length of the URLs in OnSystemRequest, StartStream and StartAudioStream and let the max size of the string be constrained by the system architecture in both the Mobile API and HMI API.
+
+## Mobile API
+
+We should remove the max length attribute of the following URL for OnSystemRequest. 
+
+```xml
+   <function name="OnSystemRequest" functionID="OnSystemRequestID" messagetype="notification" >
+        <description>
+            An asynchronous request from the system for specific data from the device or the cloud or response to a request from the device or cloud
+            Binary data can be included in hybrid part of message for some requests (such as Authentication request responses)
+        </description>
+        <param name="requestType" type="RequestType" mandatory="true">
+            <description>The type of system request.</description>
+        </param>
+        <param name="url" type="String" mandatory="false">
+            <description>
+                Optional URL for HTTP requests.
+                If blank, the binary data shall be forwarded to the app.
+                If not blank, the binary data shall be forwarded to the url with a provided timeout in seconds.
+            </description>
+        </param>
+        ...
+```   
+
+## HMI API
+
+We should remove the max length attribute of the following URL for OnSystemRequest in HMI API. 
+
+```xml
+   <function name="OnSystemRequest" messagetype="notification" >
+      <description>
+        An asynchronous request from the system for specific data from the device or the cloud or response to a request from the device or cloud
+        Binary data can be included in hybrid part of message for some requests (such as Authentication request responses)
+      </description>
+      <param name="requestType" type="Common.RequestType" mandatory="true">
+        <description>The type of system request.</description>
+      </param>
+      <param name="url" type="String" minlength="1" mandatory="false">
+        <description>Optional array of URL(s) for HTTP requests.</description>
+      </param>
+      ...
+```   
+
+The URL parameters in StartStream and StartAudioStream in HMI API should also have max length removed.
+
+```xml
+  <function name="StartStream" messagetype="request">
+    <description>Request from SmartDeviceLinkCore to HMI to start playing video streaming.</description>
+    <param name="url" type="String" minlength="21" mandatory="true">
+      <description>URL that HMI start playing.</description>
+    </param>
+    ...
+  <function name="StartAudioStream" messagetype="request">
+   <description>Request from SmartDeviceLinkCore to HMI to start playing audio streaming.</description>
+   <param name="url" type="String" minlength="21" mandatory="true">
+     <description>URL that HMI start playing.</description>
```

Comment:
I think this should be updated to "URL that HMI starts playing."

Reply:
I was wondering if I shouldn't change improper grammar here because that is what is given in https://github.com/smartdevicelink/sdl_core/blob/master/src/components/interfaces/HMI_API.xml?